### PR TITLE
Change to enable the customer creation

### PR DESCRIPTION
**Pull request by igorkosta**
Guys @razielgn, @gsamokovarov, please review.

### CODE REVIEW

File: schema/v1.0/customer.json [hunk 1]
```diff
@@ -3,26 +3,32 @@
...
     "email" : {
-      "$ref"     : "./base_types/base_types.json#definitions/email",
-      "readOnly" : true
+      "$ref"     : "./base_types/base_types.json#definitions/email"
+    },
+    "password"   : {
+      "description" : "Password for accessing your account later. You can use lower and upper case letters, digits and special characters",
+      "type"        : ["string", "null"],
+      "minLength"   : 8,
+      "pattern"     : "^[A-Za-z0-9~!@#$%^&*()_+ -]*$"
```
Comment (razielgn) [inline schema/v1.0/customer.json]: Can we change it to ```^[0-9a-zA-Z!"#$%&\'()*+,-=./:;?@\[\\\]<>^_`{|}~]+$```, which is basically all the "printable" chars from ASCII? :wink:

File: schema/v1.0/customer.json [hunk 2]
```diff
@@ -3,26 +3,32 @@
...
       "$ref" : "./base_types/base_types.json#definitions/id"
     },
     "email" : {
-      "$ref"     : "./base_types/base_types.json#definitions/email",
-      "readOnly" : true
+      "$ref"     : "./base_types/base_types.json#definitions/email"
+    },
+    "password"   : {
+      "description" : "Password for accessing your account later. You can use lower and upper case letters, digits and special characters",
+      "type"        : ["string", "null"],
```
Comment (razielgn) [inline schema/v1.0/customer.json]: Why also `null`?

File: schema/v1.0/customer.json [hunk 3]
```diff
@@ -3,26 +3,32 @@
...
+      "minLength"   : 8,
```
Comment (razielgn) [inline schema/v1.0/customer.json]: I'd add a `maxLength` too, it's `128` in our case. :smiley:

File: schema/v1.0/customer.json (hunk 2 above)
Comment (igorkosta) [inline schema/v1.0/customer.json]: It basically means, when you create the object the `password` property doesn't have to be in the response.

Comment (razielgn) [inline schema/v1.0/customer.json]: Ah, got it!

File: schema/v1.0/customer.json (hunk 1 above)
Comment (a2800276) [inline schema/v1.0/customer.json]: I'd try not to, most of fidor's customers are german and they might like "süpersecüre" as a password. I'd avoid arbitrary restrictions on password contents (might be personal: it always annoys me). Of course you have to make sure you're always getting them in a fixed encoding to handle passwords containing '☃'

Comment (razielgn) [inline schema/v1.0/customer.json]: If there isn't any constraint I don't know about, anything that's valid UTF-8 can be handled just fine. :+1:

Comment (igorkosta) [inline schema/v1.0/customer.json]: @a2800276 no one in Russia uses cyrillic for passwords. Give me the `pattern` for "süpersecüre" password and I will include it.

Comment (a2800276) [inline schema/v1.0/customer.json]: My point was more: why make the effort to implement some arbitrary restrictions concerning password content  (where we can be fairly sure we miss some special case, maybe chinese like ideogram passwords ...) , especially if we can handle anything right now.

If anything we should be checking structural integrity of passwords (min len, repeating char, variation, etc)

Added bonus: not implementing restrictions spares development resources :)

**Comment by schorsch** [comment]
Btw. are we creating a customer and a user? If so, the users params like (email) passwort should only be available in the POST endpoints params. How about errors during the user creation? e.g empty passwort is probably an error on one of our user models and not the customers one

**Comment by igorkosta** [comment]
@schorsch - but `email` should also be available on `GET /customers/{id}` which already is the case. I can move `password` to the `POST` links section.

**Comment by razielgn** [comment]
:ok_hand: thanks!

**Pull request merged**